samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaStreamSpec.java

```diff
@@ -0,0 +1,102 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.system.kafka;
+
+import java.util.Properties;
+import org.apache.samza.config.KafkaConfig;
+import org.apache.samza.system.StreamSpec;
+
+
+/**
+ * Extends StreamSpec with the ability to easily get the topic replication factor.
+ */
+public class KafkaStreamSpec extends StreamSpec {
+  private static final int DEFAULT_PARTITION_COUNT = 1;
+  private static final int DEFAULT_REPLICATION_FACTOR = 2;
+
+  /**
+   * The number of replicas for stream durability.
+   */
+  private final int replicationFactor;
+
+  /**
+   * Converts any StreamSpec to a KafkaStreamSpec.
+   * If the original spec already is a KafkaStreamSpec, it is simply returned.
+   *
+   * @param originalSpec  the StreamSpec instance to convert to KafkaStreamSpec.
+   * @return              a KafkaStreamSpec instance.
+   */
+  public static KafkaStreamSpec fromSpec(StreamSpec originalSpec) {
+    if (originalSpec instanceof KafkaStreamSpec) {
+      return ((KafkaStreamSpec) originalSpec);
+    }
+
+    int replicationFactor = Integer.parseInt(originalSpec.getOrDefault(KafkaConfig.TOPIC_REPLICATION_FACTOR(), KafkaConfig.TOPIC_DEFAULT_REPLICATION_FACTOR()));
+    return new KafkaStreamSpec(originalSpec.getId(), originalSpec.getPhysicalName(), originalSpec.getSystemName(),
+        originalSpec.getPartitionCount(), replicationFactor, originalSpec.getProperties());
+  }
+
+  /**
+   * Convenience constructor to create a KafkaStreamSpec with just a topicName, systemName, and partitionCount.
+   *
+   * @param topicName       The name of the topic.
+   * @param systemName      The name of the System. See {@link org.apache.samza.system.SystemFactory}
+   * @param partitionCount  The number of partitions.
+   */
+  public KafkaStreamSpec(String topicName, String systemName, int partitionCount) {
+    this(topicName, topicName, systemName, partitionCount, DEFAULT_REPLICATION_FACTOR, new Properties());
+  }
+
+  /**
+   * Constructs a StreamSpec with a replication factor.
+   *
+   * @param id                The application-unique logical identifier for the stream. It is used to distinguish between
+   *                          streams in a Samza application so it must be unique in the context of one deployable unit.
+   *                          It does not need to be globally unique or unique with respect to a host.
+   *
+   * @param topicName         The physical identifier for the stream. This is the identifier that will be used in remote
+   *                          systems to identify the stream. In Kafka this would be the topic name whereas in HDFS it
+   *                          might be a file URN.
+   *
+   * @param systemName        The System name on which this stream will exist. Corresponds to a named implementation of the
```

Comment:
It maybe just me. I felt it more natural to have systemName before topicName, since the scope of system is bigger. The same to the method signature in the parent class StreamSpec.